### DISCUSSION TITLE
How to use evrete as a calculation engine?

### DISCUSSION THREAD

**Discussion by shahamit** [evrete/evrete#60] [Q&A]
I am new to the evrete rule engine. I want to understand how to use it for mathematical calculations initially (in future we plan to start using it as a rule engine, currently there are no conditions - when part when executing a calculation). I have multiple financial records coming as input and I want to apply calculations over it - sum, count, avg, custom etc. **The math calculation formulae is also given by the user at runtime**

Could you please guide me on how could I define these calculations in evrete? Any reference examples would be a great start.

Thanks.

**Comment by andbi** [comment]
More info on the use case would be helpful. So far it looks like you rather need a parser of math expressions. Rule engines maintain state and perform different actions depending on their current state and incoming data.

**Reply by shahamit** [reply]
@andbi - thanks for the reply. We know that in the near future we need a rule engine hence I am trying to use evrete. For now we want to start with mathematical operations supporting sum, avg, min, max, logical etc operations.
